ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -655,11 +655,11 @@ extension ClientQueryViewController : OCQueryDelegate {
 	}
 
 	func queryHasChangesAvailable(_ query: OCQuery!) {
-		query.requestChangeSet(withFlags: OCQueryChangeSetRequestFlag(rawValue: 0)) { (_, changeSet) in
+		query.requestChangeSet(withFlags: OCQueryChangeSetRequestFlag(rawValue: 0)) { (query, changeSet) in
 			DispatchQueue.main.async {
 
-				switch query.state {
-				case .idle, .targetRemoved, .contentsFromCache, .stopped:
+				switch query?.state {
```

Comment:
Ohh nice tip. Thank you!

Reply:
On this case it does not apply because `query ` could be null but `changeSet` not. Of course if we check the SDK it should return always `query` but in my opinion we should write our code taking into account that SDK or other libraries could change